pages/admin/lessons/[lessonId]/[pageName]/index.tsx

```diff
@@ -0,0 +1,185 @@
+import { gql, useQuery } from '@apollo/client'
+import { GetAppProps, withGetApp } from '../../../../../graphql'
+import { get, sortBy, toNumber, toUpper } from 'lodash'
+import React, { useEffect, useMemo, useState } from 'react'
+import AdminLessonNav from '../../../../../components/admin/lessons/AdminLessonNav'
+import AdminLessonSideNav from '../../../../../components/admin/lessons/AdminLessonSideNav'
+import AdminModuleInputs, {
+  Props
+} from '../../../../../components/admin/lessons/AdminModuleInputs'
+import Breadcrumbs from '../../../../../components/Breadcrumbs'
+import styles from '../../../../../scss/modules.module.scss'
+import navStyles from '../../../../../scss/adminLessonNav.module.scss'
+import { AdminLayout } from '../../../../../components/admin/AdminLayout'
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+
+const MODULES = gql`
+  query {
+    modules {
+      id
+      name
+      content
+      lesson {
+        id
+      }
+      order
+    }
+  }
+`
+
+type Module = {
+  id: number
+  name: string
+  content: string
+  lesson: { id?: number }
+  order: number
+}
+type Modules = Module[]
+
+type ModulesTabProps = {
+  filteredModules: Modules
+  lessonId: number
+  refetch: Props['refetch']
+}
+
+const ModulesTab = ({
+  filteredModules,
+  lessonId,
+  refetch
+}: ModulesTabProps) => {
+  const [selectedIndex, setSelectedIndex] = useState(-1)
+  const onAddItem = () => setSelectedIndex(-1)
+  const onSelect = (item: Omit<Module, 'order'>) => setSelectedIndex(item.id)
+
+  useEffect(() => setSelectedIndex(get(filteredModules[0], 'id')), [lessonId])
+
+  return (
+    <div className={styles.container__modulesPanel}>
+      <AdminLessonSideNav
+        title="Modules"
+        items={filteredModules}
+        onAddItem={onAddItem}
+        onSelect={onSelect}
+        selectedIndex={selectedIndex}
+      />
+      <div className={styles.container__modulesPanel__inputs}>
+        <AdminModuleInputs
+          lessonId={lessonId}
+          refetch={refetch}
+          module={filteredModules.find(module => module.id === selectedIndex)}
+        />
+      </div>
+    </div>
+  )
+}
+
+const Lessons = ({ data }: GetAppProps) => {
+  const router = useRouter()
+  const { pageName, lessonId } = router.query
+
+  const { lessons } = data
+  const { data: modulesData, refetch } = useQuery<{ modules: Modules }>(MODULES)
+
+  const [lesson, setLesson] = useState({
+    title: '',
+    id: -1
+  })
+  const [modules, setModules] = useState(
+    sortBy(get(modulesData, 'modules'), 'order')
+  )
+
+  const filteredModules = useMemo(
+    () => modules.filter(module => module.lesson.id === lesson.id),
+    [lesson.id, modules]
+  )
+
+  useEffect(() => {
+    setModules(sortBy(get(modulesData, 'modules'), 'order'))
+  }, [get(modulesData, 'modules')])
+
+  useEffect(() => {
```

Comment:
why does this need to be inside `useEffect`? If lesson data is passed in as prop and lessonId is retrieved in the url, you can just get current lesson right away

Reply:
I agree. `lessons` should be out of the `useEffect` dependencies, but the `lessonId` has to stay because in the first couple of renders, `lessonId` is undefined. It seems like `useRouter` doesn't immediately extract the data from the URL.

Reply:
I removed `lessons` but it seems it doesn't resolve immediately. It has to exist